engines/native/sandbox.go

```diff
@@ -16,59 +16,77 @@ import (
 
 type sandbox struct {
 	engines.SandboxBase
-	engine     *engine
-	context    *runtime.TaskContext
-	log        *logrus.Entry
-	homeFolder runtime.TemporaryFolder
-	user       *system.User
-	process    *system.Process
-	env        map[string]string
-	resolve    atomics.Once // Guarding resultSet, resultErr and abortErr
-	resultSet  *resultSet
-	resultErr  error
-	abortErr   error
-	wg         atomics.WaitGroup
+	engine        *engine
+	context       *runtime.TaskContext
+	log           *logrus.Entry
+	workingFolder runtime.TemporaryFolder
+	user          *system.User
+	process       *system.Process
+	env           map[string]string
+	resolve       atomics.Once // Guarding resultSet, resultErr and abortErr
+	resultSet     *resultSet
+	resultErr     error
+	abortErr      error
+	wg            atomics.WaitGroup
 }
 
 func newSandbox(b *sandboxBuilder) (*sandbox, error) {
 	// Create temporary home folder for the task
-	homeFolder, err := b.engine.environment.TemporaryStorage.NewFolder()
+	workingFolder, err := b.engine.environment.TemporaryStorage.NewFolder()
 	if err != nil {
 		b.log.Error("Failed to create temporary folder: ", err)
 		return nil, fmt.Errorf("Failed to temporary folder, error: %s", err)
 	}
 
 	if b.payload.Context != "" {
-		if err = fetchContext(b.payload.Context, homeFolder.Path()); err != nil {
+		if err = fetchContext(b.payload.Context, workingFolder.Path()); err != nil {
 			b.context.LogError(err)
 			return nil, engines.NewMalformedPayloadError(
 				fmt.Sprintf("Error downloading %s: %v", b.payload.Context, err),
 			)
 		}
 	}
 
-	// Create temporary user account
-	user, err := system.CreateUser(homeFolder.Path(), b.engine.groups)
-	if err != nil {
-		homeFolder.Remove() // best-effort clean-up this is a fatal error
-		return nil, fmt.Errorf("Failed to create temporary system user, error: %s", err)
+	var user *system.User
+	var username string
+
+	if b.engine.config.CreateUser {
+		// Create temporary user account
+		user, err = system.CreateUser(workingFolder.Path(), b.engine.groups)
+		if err != nil {
+			workingFolder.Remove() // best-effort clean-up this is a fatal error
+			return nil, fmt.Errorf("Failed to create temporary system user, error: %s", err)
+		}
+		username = user.Name()
+	} else {
+		var curUser *system.User
+
+		user = nil
+
+		curUser, err = system.CurrentUser()
```

Comment:
Can't we modify `system.StartProcess(system.ProcessOptions{` to accept `Owner:  system.CurrentUser()` ?

That way this becomes:
```go
var user *system.User
if b.engine.config.CreateUser {
  user, err = system.CreateUser(workingFolder.Path(), b.engine.groups)
} else {
  user, err = system.CurrentUser()
}
if (err != nil) {
  ....
}

env["LOGNAME"] = user.Name()
```

It would reduce a lot of special cases, and seems rather reasonable that `Owner` could be the user object returned from `CurrentUser()`.